src/tools.js

```diff
@@ -622,7 +622,9 @@ window.tools = [{
 				name: "Curve",
 				icon: get_icon_for_tool(this),
 			}, ()=> {
-				this.draw_curve(ctx);
+				if (this.points.length >= 1) {
```

Comment:
You can remove the condition in this case since it's finishing the curve here, so it's guaranteed to have 4 points.

Reply:
fixed. I amended to the previous commit since it was a small change.